src/helper-modules/govuk-notify/govuk-notify.service.test.ts

```diff
@@ -0,0 +1,140 @@
+      expect(response).toEqual({ status: 201, data: expectedResponse });
+    });
+
+    it.each([
+      {
+        exceptionClass: BadRequestException,
+        exceptionName: 'Bad Request Exception',
+        error: 'Bad Request',
+        status: 400,
+      },
+      {
+        exceptionClass: UnauthorizedException,
+        exceptionName: 'Unauthorized Exception',
+        error: 'Unauthorized',
+        status: 401,
+      },
+      {
+        exceptionClass: ForbiddenException,
+        exceptionName: 'Forbidden Exception',
+        error: 'Forbidden',
+        status: 403,
+      },
+      {
+        exceptionClass: InternalServerErrorException,
+        exceptionName: 'Internal Server Error Exception',
+        error: 'Internal Server Error',
+        status: 500,
+      },
+    ])('throws exception $exceptionName for unexpected $status', async ({ exceptionClass, exceptionName, error, status }) => {
+      jest.mocked(sendEmailMethodMock).mockImplementation(() => Promise.reject(generateNotifyError(status, errorMessage)));
+
+      const resultPromise = service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation });
+
+      expect(sendEmailMethodMock).toHaveBeenCalledTimes(1);
+      await expect(resultPromise).rejects.toBeInstanceOf(exceptionClass);
+      await expect(resultPromise).rejects.toThrow(exceptionName);
+      await expect(resultPromise).rejects.toHaveProperty('status', status);
+      await expect(resultPromise).rejects.toHaveProperty('response', { message: [errorMessage], error, statusCode: status });
+    });
+
+    it('throws generic Error exception for unexpected status', async () => {
+      const unexpectedStatus = valueGenerator.integer({ min: 900, max: 999 });
+      jest.mocked(sendEmailMethodMock).mockImplementation(() => Promise.reject(generateNotifyError(unexpectedStatus, errorMessage)));
+
+      const resultPromise = service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation });
+
+      expect(sendEmailMethodMock).toHaveBeenCalledTimes(1);
+      await expect(resultPromise).rejects.toBeInstanceOf(Error);
+      await expect(resultPromise).rejects.toThrow(errorMessage);
+      await expect(resultPromise).rejects.toHaveProperty('message', errorMessage);
+      await expect(resultPromise).rejects.toHaveProperty('stack');
+    });
+
+    it('throws exception UnprocessableEntityException for empty response from GOV.UK Notify client', async () => {
+      jest.mocked(sendEmailMethodMock).mockImplementation(() => Promise.resolve(''));
+
+      const resultPromise = service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation });
+
+      expect(sendEmailMethodMock).toHaveBeenCalledTimes(1);
+      await expect(resultPromise).rejects.toBeInstanceOf(UnprocessableEntityException);
+      await expect(resultPromise).rejects.toThrow('No GOV.UK Notify response');
+    });
+  });
```

Comment:
I tried, but found checking logger call after exception a bit tricky.

We don't test for logger call in MDM API, so it should be fine to avoid this test here.